### PR TITLE
Implement Renovate

### PR DESCRIPTION
**Pull request by hcsa73**
Removed checksum verification (can't be done with Renovate)

### CODE REVIEW

**Review by vicentepinto98 — CHANGES_REQUESTED** [review]
I guess it would also make sense to somehow tell explicitly that the generation should be run after this update, because the possible changes/errors should be dealt with directly there and not pop up with the next SDK generation. What about adding `make generate-sdk` as a postUpdateCommand?

**Comment by hcsa73** [comment]
> I guess it would also make sense to somehow tell explicitly that the generation should be run after this update, because the possible changes/errors should be dealt with directly there and not pop up with the next SDK generation. What about adding `make generate-sdk` as a postUpdateCommand?

It seems best to add a CI pipeline that generates/lints/tests the SDK, since this is not Renovate-specific

File: scripts/generate-sdk/generate-sdk.sh [hunk 1]
```diff
@@ -15,6 +13,10 @@ SDK_REPO="https://github.com/stackitcloud/stackit-sdk-go.git"
 SDK_GO_VERSION="1.18"
 OAS_REPO=https://github.com/stackitcloud/stackit-api-specifications
 
+# Renovate: datasource=github-tags depName=OpenAPITools/openapi-generator versioning=regex:^(v)?(?<version>[0-9]+\.[0-9]+\.[0-9]+)$
+GENERATOR_VERSION="v6.5.0"
+GENERATOR_VERSION_NUMBER="${GENERATOR_VERSION:1}"
```
[the remark this answers is not in the document]
Comment (hcsa73) [reply scripts/generate-sdk/generate-sdk.sh:18]: Capture groups needed for Renovate to work. Will try to simplify them

File: scripts/generate-sdk/generate-sdk.sh [hunk 2]
```diff
@@ -56,17 +58,13 @@ if [ ! -d ${ROOT_DIR}/oas ]; then
...
-if [ -e ${jar_path} ] && [ $(java -jar ${jar_path} version) == ${GENERATOR_VERSION} ] && [ $(shasum -a 512 ${jar_path} | awk '{print $1;}') == ${GENERATOR_SHASUM_512} ]; then
+if [ -e ${jar_path} ] && [ $(java -jar ${jar_path} version) == ${GENERATOR_VERSION_NUMBER} ]; then
     :
 else
     echo "Downloading OpenAPI generator (version ${GENERATOR_VERSION}) to ${GENERATOR_PATH}..."
     mkdir -p ${GENERATOR_PATH}
-    wget https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/${GENERATOR_VERSION}/openapi-generator-cli-${GENERATOR_VERSION}.jar -O ${jar_path}
+    wget https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/${GENERATOR_VERSION_NUMBER}/openapi-generator-cli-${GENERATOR_VERSION_NUMBER}.jar -O ${jar_path}
     echo "Download done."
-    if [ $(shasum -a 512 ${jar_path} | awk '{print $1;}') != ${GENERATOR_SHASUM_512} ]; then
```
[the remark this answers is not in the document]
Comment (hcsa73) [reply scripts/generate-sdk/generate-sdk.sh:66]: Requires someone to pull the branch, download the binary, compute the check sum and update the variable. Not sure it's worth it, but could be done

[the remark this answers is not in the document]
Comment (vicentepinto98) [reply scripts/generate-sdk/generate-sdk.sh:66]: How did we get the checksum that is being used currently?

[the remark this answers is not in the document]
Comment (hcsa73) [reply scripts/generate-sdk/generate-sdk.sh:66]: By downloading the binary and computing the checksum locally

[the remark this answers is not in the document]
Comment (vicentepinto98) [reply scripts/generate-sdk/generate-sdk.sh:66]: I thought there was a way to get it from the website. Then this is not that relevant, can be removed

**Review by vicentepinto98 — DISMISSED** [review]

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**